### PR TITLE
Update docs for new TileMapLayer node

### PR DESCRIPTION
**Pull request by skyace65**
Does what the title says. The TileMapLayer node has been added to 4.3 and TileMap is now marked as deprecated. I've gone through and updated every page I could find with a TileMap reference, not just  the two main pages. screenshots have been updated and gone through Squoosh compression.

### CODE REVIEW

File: tutorials/2d/using_tilemaps.rst [hunk 1]
```diff
@@ -400,7 +403,7 @@ set for the TileSet yet.
 There are 3 kinds of painting modes available for terrain connections:
 
 - **Connect**, where tiles are connected to surrounding tiles on the same
-  TileMap layer.
+  TileMap layer. #CHECK LATER
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:406]: Is this supposed to be a comment? In rst that should be:
```suggestion
  TileMap layer. .. CHECK LATER
```

Comment (skyace65) [reply tutorials/2d/using_tilemaps.rst]: Actually I left that for myself as a reminder to review that section, forgot to do that, just updated the PR.

File: tutorials/2d/using_tilemaps.rst [hunk 2]
```diff
@@ -39,48 +39,51 @@ resource and choose **Save**:
...
-
-.. figure:: img/using_tilemaps_create_layers.webp
-   :align: center
-   :alt: Creating layers in a TileMap node (example with "background" and "foreground")
+Multiple TileMaplayers and settings
+-----------------------------------
 
-   Creating layers in a TileMap node (example with "background" and "foreground")
+When working with tilemaps it's generally advised that you use multiple TileMapLayer
+nodes when appropriate. Using multiple layers can be adventageuous, for example,
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:46]:
```suggestion
nodes when appropriate. Using multiple layers can be advantageous, for example,
```

File: tutorials/2d/using_tilemaps.rst [hunk 3]
```diff
@@ -39,48 +39,51 @@ resource and choose **Save**:
...
+- **TileSet** The tileset used by the TileMapLayer node.
+
+Rendering
+^^^^^^^^^
 
-You can reorder layers by drag-and-dropping the "three horizontal bars" icon on
-the left of the entries in the **Layers** section.
+- **Y Sort Origin:** The vertical offset to use for Y-sorting on each tile (in pixels).
+  Only effective if **Y Sort Enabled** under CanvasItem settings is ``true``.
+- **X Draw Order Reversed** Reverses the order tiles are drawn on the X axis. requires
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:62]:
```suggestion
- **X Draw Order Reversed** Reverses the order tiles are drawn on the X axis. Requires that
```

File: tutorials/2d/using_tilemaps.rst [hunk 4]
```diff
@@ -39,48 +39,51 @@ resource and choose **Save**:
...
+  **Y Sort Enabled** under CanvasItem settings is ``true``.
+- **Rendering Quadrant Size** A quadrant is a group of tiles drawn together on a single
+  CanvasItem for optimization purposes. This setting defines the length of a squares
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:65]:
```suggestion
  CanvasItem for optimization purposes. This setting defines the length of a square's
```

File: tutorials/2d/using_tilemaps.rst [hunk 5]
```diff
@@ -39,48 +39,51 @@ resource and choose **Save**:
...
+- **Collision Enabled** Enables or disables collision.
+- **Use Kinematic Bodies** When true TileMapLayer collision shapes will be instantiated
+  as kinematic bodies.
+- **Collision Visibility Mode** Whether or not the TileMapLayer's collision shapes are
+  visible. If set to default, then it depends on the show collision debug settings.
+
+Navigation
+^^^^^^^^^^
+
+- **Navigation Enabled** Whether or not navigation regions are enabled
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:80]:
```suggestion
- **Navigation Enabled** Whether or not navigation regions are enabled.
```

**Review by AThousandShips — APPROVED** [review]
Looks good style wise

**Comment by skyace65** [comment]
Fixed.

File: tutorials/2d/using_tilemaps.rst [hunk 6]
```diff
@@ -39,48 +39,51 @@ resource and choose **Save**:
...
+- **TileSet** The tileset used by the TileMapLayer node.
+
+Rendering
+^^^^^^^^^
 
-You can reorder layers by drag-and-dropping the "three horizontal bars" icon on
-the left of the entries in the **Layers** section.
+- **Y Sort Origin:** The vertical offset to use for Y-sorting on each tile (in pixels).
+  Only effective if **Y Sort Enabled** under CanvasItem settings is ``true``.
+- **X Draw Order Reversed** Reverses the order tiles are drawn on the X axis. requires
```
Comment (AThousandShips) [inline tutorials/2d/using_tilemaps.rst:62]:
```suggestion
- **X Draw Order Reversed** Reverses the order tiles are drawn on the X axis. Requires
```

**Comment by skyace65** [comment]
Fixed again.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Great work! Merged.

**Pull request merged**